service/history/queueProcessor.go

```diff
@@ -267,93 +267,116 @@ func (p *queueProcessorBase) retryTasks() {
-			logging.LogCriticalErrorEvent(logger, "Critical error processing transfer task.  Skipping.", err)
-			p.metricsClient.IncCounter(p.options.MetricScope, metrics.CadenceCriticalFailures)
-			p.ackMgr.completeQueueTask(task.GetTaskID())
-			return
-		}
+func (p *queueProcessorBase) processTaskOnce(notificationChan <-chan struct{}, task queueTaskInfo, logger bark.Logger) (int, error) {
 
-		logging.LogOperationPanicEvent(logger, "Retry count exceeded for transfer task", err)
-	case *persistence.ReplicationTaskInfo:
-		// Cannot processes replication task due to LimitExceededError after all retries
-		// raise and alert and move on
-		if _, ok := err.(*workflow.LimitExceededError); ok {
-			logging.LogCriticalErrorEvent(logger, "Critical error processing replication task.  Skipping.", err)
-			p.metricsClient.IncCounter(p.options.MetricScope, metrics.CadenceCriticalFailures)
-			p.ackMgr.completeQueueTask(task.GetTaskID())
-			return
+	select {
+	case <-notificationChan:
+	default:
+	}
+
+	startTime := time.Now()
+	scope, err := p.processor.process(task)
+	p.metricsClient.IncCounter(scope, metrics.TaskRequests)
+	p.metricsClient.RecordTimer(scope, metrics.TaskProcessingLatency, time.Since(startTime))
+
+	return scope, err
+}
+
+func (p *queueProcessorBase) handleTaskError(scope int, startTime time.Time,
+	notificationChan <-chan struct{}, err error, logger bark.Logger) error {
+
+	if _, ok := err.(*workflow.EntityNotExistsError); ok {
+		return nil
+	}
+
+	// this is a transient error
+	if err == ErrTaskRetry {
+		<-notificationChan
+		return err
+	}
+
+	// this is a transient error
+	if _, ok := err.(*workflow.DomainNotActiveError); ok {
+		if time.Now().Sub(startTime) > cache.DomainCacheRefreshInterval {
+			p.metricsClient.IncCounter(scope, metrics.TaskNotActiveCounter)
+			return nil
 		}
 
-		logging.LogOperationPanicEvent(logger, "Retry count exceeded for replication task", err)
+		return err
+	}
+
+	p.metricsClient.IncCounter(scope, metrics.TaskFailures)
+
+	if _, ok := err.(*persistence.CurrentWorkflowConditionFailedError); ok {
+		logging.LogTaskProcessingFailedEvent(logger, "More than 2 workflow is running.", err)
+		return nil
 	}
-}
 
-func (p *queueProcessorBase) initializeLoggerForTask(task queueTaskInfo, logger bark.Logger) bark.Logger {
-	if logger != nil {
-		return logger
+	if _, ok := err.(*workflow.LimitExceededError); ok {
+		p.metricsClient.IncCounter(scope, metrics.TaskLimitExceededCounter)
+		logging.LogTaskProcessingFailedEvent(logger, "Task encounter limit exceeded error.", err)
+		return err
 	}
 
-	logger = p.logger.WithFields(bark.Fields{
-		logging.TagTaskID:   task.GetTaskID(),
-		logging.TagTaskType: task.GetTaskType(),
-		logging.TagVersion:  task.GetVersion(),
+	logging.LogTaskProcessingFailedEvent(logger, "Fail to process task", err)
+	return err
+}
+
+func (p *queueProcessorBase) ackTaskOnce(task queueTaskInfo, scope int) {
+	p.ackMgr.completeQueueTask(task.GetTaskID())
```

Comment:
Looks like missing err handling for completeQueueTask

Reply:
this is intentional, will apply range delete replication task. after that, this issue is gone